### PR TITLE
Documenting exposed commands

### LINKED ISSUE #1360: Generate a list of all lazy commands
See #1358 for rationale.

We should have all the available lazy commands listed in the documentation, either on the ["Lazy objects"](http://docs.qtile.org/en/latest/manual/config/lazy.html) or directly along the resource that provide them (for example on the ["Built-in Layouts"](http://docs.qtile.org/en/latest/manual/ref/layouts.html) page).

### PR DESCRIPTION
**Pull request by elParaguayo**
This PR adds a comprehensive list of all commands exposed via IPC i.e. those that can be accessed by `lazy`, `qtile cmd-obj` etc.

Draft for now as it's messy but it's worth having a chat about the output.

@m-col - I feel like we may have discussed this already...

Commands can be viewed here: https://qtile--3540.org.readthedocs.build/en/3540/manual/ref/commands.html  (NB the link will die after a certain amount of time - if it does, you'll need to fetch the PR and build locally).

Closes #1360

### CODE REVIEW

File: docs/manual/ref/commands.rst
Comment (m-col) [reply docs/manual/ref/commands.rst:12]: On the RTD page these aren't being rendered as bullet points btw

**Review by m-col — COMMENTED** [review]
Overall +1 from me, this is great and I think in the right direction.

I think eventually we can merge this tab with the 'Scripting' and 'Commands API' tabs, having dedicated sections for each of the interfaces through which commands can be accessed. So I think the very top of this page that lists the interfaces doesn't really need them, so this becomes an interface-agnostic commands reference. That said it has to be clear how each object is named at each point (i think the only not-so-obvious one is the root, via `qtile cmd-obj -o cmd`), but the rest are consistent across interfaces and so that will keep things clear.

Similarly we might be able to drop the `lazy.X` paths in the first column of each table. This would also help with layout of the page because currently the argument column is quite thin.

Also re the window classes, it should (in theory) be sufficient to show the commands on `backend.base.Window`. The XWindow/XdgWindow distinction, for example, is an implementation detail that the user's don't need to (and shouldn't) be aware of. The only backend-specific window commands are `cmd_hints` and `cmd_inspect` on the X11 window class, but they are both for debugging, so maybe we can get away without documenting those? Maybe with the command decorators we can be more explicit with that by using leading underscores on their command names.

Lastly I noticed many command docstrings got parsed poorly so we'll have to go through and make all of them consistent and nice. I'm happy to help out with that.

**Comment by elParaguayo** [comment]
The other bit we should probably be adding here is some bits about qtile's internals e.g. when people write their own `lazy` functions, they need to access methods and attributes of the `qtile` object.

File: docs/manual/ref/commands.rst
Comment (elParaguayo) [reply docs/manual/ref/commands.rst:12]: > "or scripting commands to qtile." - what does this mean?

I've forgotten! I wrote this a while ago. It's certainly not clear.

> Is this list of interfaces comprehensive itself? That might also be nice. I think the ones not yet here are the Python interface (is that what you meant with scripting commands?), and perhaps custom functions that access them directly.

No, not comprehensive. As you say, there's also the `CommandClient` and `InteractiveCommandClient`.

Comment (elParaguayo) [reply docs/manual/ref/commands.rst:12]: > On the RTD page these aren't being rendered as bullet points btw

Odd. Bullets show when building locally.

**Comment by elParaguayo** [comment]
> Lastly I noticed many command docstrings got parsed poorly so we'll have to go through and make all of them consistent and nice. I'm happy to help out with that.

This is actually causing me a fair bit of grief.

The issue is that some of the commands have `numpydoc` style docstring which includes some rst style formatting (section headings and code blocks). To get the text into a table I had to treat this all as a block of text which is why it looks ugly.

The alternative is to use the `.. method::` directive and show the docstring with its format. That would look like this:


Personally, I don't think this looks as neat as the table but it does make the docstring much clearer.

If we want to have the docstring in the table then I suspect they do need to be rewritten.

**Comment by m-col** [comment]
I think a more comprehensible approach would be to have the tables display the command name and the header line from the docstring for each command. Then, if you click the command, it takes you to more full docs for that command. An example of this is these docs: https://stk.readthedocs.io/en/stable/stk.molecular.molecules.building_block.html. The methods table at the top list the name and description and its neat because it's only 2 columns. It acts almost as an index to find what you want. Then, to find out _how_ to use the command you want once you've found it, clicking it takes you to the full docstring with parameters.

This still leaves the issue of docstring formatting. Comparing numpy and google formats, google's format is more flexible because numpy requires return values and types. Ideally, all user-facing commands would have full docstrings with summary lines and parameter lists all in the same format. I'm happy to submit updated docstrings in a PR if you'd like? Then if you rebased this on top you may find this easier.

**Comment by elParaguayo** [comment]
That would be amazing if you're happy to do that.

I agree that google style looks good and the example you gave looks nice too.

**Comment by elParaguayo** [comment]
That's fine - as long as the first line is still a good summary that we can extract to replicate that example in your earlier post.

**Comment by elParaguayo** [comment]
If we want to do the `autoclass`/`automethod` directive route then we should hold this PR until after the decorators PR gets merged, otherwise the commands are going to be shown with their `cmd_` prefix which is what I was looking to avoid here.

**Comment by elParaguayo** [comment]
@m-col Now that we've merged decorators we should really get this one done. Any preferences for how we want to proceed?

**Comment by m-col** [comment]
I was just checking out that example link I mentioned above to see again how they did it (https://stk.readthedocs.io/en/stable/stk.molecular.molecules.building_block.html). There are actually lots of things I like about their docs :thinking: E.g. the navigation sidebar has sections with headers - might be nice to have a commands section (and other sections to split up our pages).

Anyway back on topic - the "Building Block" page has a class, description, table summarising methods, then detailed docs per method. We could do it similarly for commands (ignoring other methods). And each class would have its own page to make navigation easier (e.g. their following page is "Constructed Molecule" which follows the same structure for another class). If possible, collapsing classes from backends would keep things a bit neater too e.g. we'd have a generic `Window` page with its commands and docs, and we'd annotate "X11 only"/"Wayland only" where appropriate.

What do you think?

**Comment by elParaguayo** [comment]
The page does look good. Two comments:
- This means some rewriting of the docstrings to get them in to the correct format. Nothing wrong in that.
- Are you expecting users to say "I want a keybinding to float a window so I'll go to the Window page and find the command"? If not, we probably still need a page which lists the lazy commands.

**Comment by m-col** [comment]
> This means some rewriting of the docstrings to get them in to the correct format. Nothing wrong in that.

Would it be easier to get this done before or after setting up the docs to build in this way?

> Are you expecting users to say "I want a keybinding to float a window so I'll go to the Window page and find the command"? If not, we probably still need a page which lists the lazy commands.

Yep exactly. In the top docstring for the object we could have a quick example for accessing that object through each interface (python direct, python lazy, IPC)

**Comment by elParaguayo** [comment]
> > This means some rewriting of the docstrings to get them in to the correct format. Nothing wrong in that.
> 
> Would it be easier to get this done before or after setting up the docs to build in this way?

I think they happen at the same time (ie same PR) as we will need the docstrings in the right format to check the docs work.

**Comment by m-col** [comment]
Makes sense. As mentioned above I'm happy to do the tedious docstring formatting or contribute however else to this :) Perhaps if you set up the doc build with a first commit, I can append a commit fixing the docstrings?

**Comment by elParaguayo** [comment]
That works for me.

**Comment by elParaguayo** [comment]
> Anyway back on topic - the "Building Block" page has a class, description, table summarising methods, then detailed docs per method. We could do it similarly for commands (ignoring other methods). And each class would have its own page to make navigation easier (e.g. their following page is "Constructed Molecule" which follows the same structure for another class). If possible, collapsing classes from backends would keep things a bit neater too e.g. we'd have a generic `Window` page with its commands and docs, and we'd annotate "X11 only"/"Wayland only" where appropriate.

The issue I'm having with this is that, while I can produce the summary list and full detailed list for a module (like your example), there's no way (or not one that I've found) to filter out exposed commands in just the short summary but keep them in the full list.

**Comment by elParaguayo** [comment]
OK - I might start a new PR as, having started on this, I really want to refactor the docs and there's more than could be changed rather than just documenting commands better.

**Comment by m-col** [comment]
Sounds good.

> If we're trying to help users by providing more docs on the qtile attributes, methods etc then it would be good to include those in the docs.

Yeah definitely. Perhaps we can put docs for these things in the docstrings of the respective graph node if that makes it makes it more straight forward for doc generation. I don't think we'd need to include attributes beyond graph paths, nor methods that aren't commands.

**Comment by elParaguayo** [comment]
> Sounds good.
> 
> > If we're trying to help users by providing more docs on the qtile attributes, methods etc then it would be good to include those in the docs.
> 
> Yeah definitely. Perhaps we can put docs for these things in the docstrings of the respective graph node if that makes it makes it more straight forward for doc generation. I don't think we'd need to include attributes beyond graph paths, nor methods that aren't commands.

My concern was that the docstring, screenshots, parameters, commands summary and fuller documentation is too much for one page so I wonder if there's a sensible way of splitting that otherwise the widgets page will be huuuuge!

As for excluding non-command methods, that feels right and, actually does make things easier. However, there are things like `widgets_map` etc. which we should explain better but these could just be included in the main `Qtile` docstring.

**Comment by m-col** [comment]
> My concern was that the docstring, screenshots, parameters, commands summary and fuller documentation is too much for one page so I wonder if there's a sensible way of splitting that otherwise the widgets page will be huuuuge!

One page per graph node? Possibly including generic pages for nodes that have various instances like layouts, widgets. Is that helpful? Or just one page per class definition. We can make more use of nested pages in the RTD theme, so it'd be a widget section then a bunch of widget pages. Same for layouts. We could probably auto-generate those pages as part of the build step too, if they just follow a template for the rst directive.

> However, there are things like widgets_map etc. which we should explain better but these could just be included in the main Qtile docstring.

Yeah, could do. Probably best leaving this for a future change? We'd probably need to discuss further what attributes we're documenting because we'd essentially be putting those forward as 'official' API.

**Review by m-col — COMMENTED** [review]
This is really brilliant. I've made a couple ideas and suggestions inline, and here are a bunch more that don't apply specifically to your changes. I generally used this as an opportunity to review the docs so not all pertain the exposing the commands. Apologies for a bit of brain dump!

---

As you mentioned on discord there is the issue of having 2 pages serving as references for widgets and layouts, one showing configuration options and one showing commands as part of the commands API. I'm not sure what would be best here! I think I'd be in favour of keeping them separate (having 2 pages) as otherwise having 1 with both commands and config options would be big and more difficult to navigate. Having collapsable sections might cause problems with ctrl-f searching, plus configuration options and commands are different scopes. But maybe we can link to the other page at the top of each one? Or link to each widget's command section below where we render its docstring, above the parameter table.

---

Could we drop the `genindex` at the bottom of the home page? I'm thinking that the docs are already searchable using the search bar at the top, so linking to the index doesn't add anything.

---

On the 'Troubleshooting' page, should we move the two 'Common Issues' to the FAQ? This would then leave Troubleshooting as a guide for how to solve problems, rather than some specific problems. Then could we move 'Wayland' above 'Troubleshooting'. Maybe we should also more the 'Configuration' page to the top of the sidebar 'REFERENCE' section, which we could rename to 'CONFIGURATION', then rename the 'Configuration' page to 'Introduction' or something. Does that make sense?

---

What would you think about moving the 'How to create a widget' and 'Using git' pages to the 'Miscellaneous' section? It's only a superficial change but makes the sidebar shorter and appear less cluttered.

Could you add this diff to update the wayland page?
<details>

```diff
diff --git a/docs/manual/wayland.rst b/docs/manual/wayland.rst
index a521e278..105ef5f8 100644
--- a/docs/manual/wayland.rst
+++ b/docs/manual/wayland.rst
@@ -6,8 +6,10 @@ Running Qtile as a Wayland Compositor
 
 Some functionality may not yet be implemented in the Wayland compositor. Please
 see the discussion `here <https://github.com/qtile/qtile/discussions/2409>`_ to
-see the current state of development. Also see :ref:`troubleshooting
-<debugging-wayland>`.
+see the current state of development. See `here
+<https://github.com/qtile/qtile/labels/Wayland>`_ for unresolved
+Wayland-specific issues and see :ref:`troubleshooting <debugging-wayland>` for
+tips on how to debug Wayland problems.
 
 Backend-Specific Configuration
 ==============================
@@ -28,9 +30,9 @@ backend in your config as follows:
 Running X11-Only Programs
 =========================
 
-Qtile _does_ support XWayland. This requires that `wlroots` and `pywlroots`
-were built with XWayland support, and that XWayland is installed on the system
-from startup. XWayland will be started the first time it is needed.
+Qtile supports XWayland but requires that `wlroots` and `pywlroots` were built
+with XWayland support, and that XWayland is installed on the system from
+startup. XWayland will be started the first time it is needed.
 
 XWayland windows sometimes don't receive mouse events
 -----------------------------------------------------

```

</details>

---

Perhaps we should move the `Shell commands` sidebar entry to the `GETTING STARTED` section, and maybe renamed to something like `Entrypoints` to distinguish it from e.g. getting qtile to executed shell commands (i.e. lazy.spawn(), which seems to more often discussed than the various entrypoints).

---

Should we move the `Scripting` section above the `Reference` section? `Scripting` seems to give context and explanation, then `Reference` makes sense as a 'so here are the tools you can use for that'. Also should we re-order the pages in the `Reference` section in order of relative importance? I'm thinking:

1. Default Config File (and might be worth adding a note to the top of the page saying this is packaged with your installation, and is copied to `~/.config/qtile/config.py` the first time you start Qtile.)
2. Layouts
3. Widgets
4. Hooks
5. Extensions

---

Perhaps we should indirectly note widget mirroring, but then exclude `Mirror` from the widgets reference page?:
<details>

```diff
diff --git a/docs/manual/config/screens.rst b/docs/manual/config/screens.rst
index 53bfc30b..89702d87 100644
--- a/docs/manual/config/screens.rst
+++ b/docs/manual/config/screens.rst
@@ -16,21 +16,28 @@ Tying together screens, bars and widgets, we get something like this:
     from libqtile.config import Screen
     from libqtile import bar, widget
 
+    window_name = widget.WindowName()
+
     screens = [
         Screen(
             bottom=bar.Bar([
                 widget.GroupBox(),
-                widget.WindowName()
+                window_name,
                 ], 30),
             ),
         Screen(
             bottom=bar.Bar([
                 widget.GroupBox(),
-                widget.WindowName()
+                window_name,
                 ], 30),
             )
         ]
 
+Note that a widget can be passed to multiple bars (and likewise multiple times
+to the same bar). Its contents is mirrored across all copies so this is useful
+where you want identical content (e.g. the name of the focussed window, like in
+this example).
+
 Bars support both solid background colors and gradients by supplying a list of
 colors that make up a linear gradient. For example, :code:`bar.Bar(...,
 background="#000000")` will give you a black back ground (the default), while
```

</details>

---

Also this is a small update for the Configuration page to note the wayland version of the xephyr script:
<details>

```diff
diff --git a/docs/manual/config/index.rst b/docs/manual/config/index.rst
index 8918e30e..b51c5d97 100644
--- a/docs/manual/config/index.rst
+++ b/docs/manual/config/index.rst
@@ -184,11 +184,13 @@ configuration variables that control specific aspects of Qtile's behavior:
 Testing your configuration
 ==========================
 
-The best way to test changes to your configuration is with the provided Xephyr
-script. This will run Qtile with your ``config.py`` inside a nested X server
-and prevent your running instance of Qtile from crashing if something goes
-wrong.
+The best way to test changes to your configuration is with the provided scripts
+at `./scripts/xephyr`_ (X11) or `./scripts/wephyr`_ (Wayland). This will run
+Qtile with your ``config.py`` inside a nested window and prevent your running
+instance of Qtile from crashing if something goes wrong.
+
+.. _./scripts/xephyr: https://github.com/qtile/qtile/blob/master/scripts/xephyr
+.. _./scripts/wephyr: https://github.com/qtile/qtile/blob/master/scripts/wephyr
 
 See :ref:`Hacking Qtile <hacking>` for more information on using
 Xephyr.
-

```

</details>

---

Your 'Navigating the command graph' page is a nice improvement over the previous 'Keys' one. Though I think not having it nested within 'Introduction: Object Graph' would make navigating that part of the docs more clear. And should we make the 'Client-Server Scripting Model' page, with its example,  a page under the 'Commands API' sidebar section? To summary, 'Commands API' would look like:

- Introduction: Object Graph
- Navigating the graph
- Client-Server Scripting Model (example would be on this page but not in the TOC)
- Object graph development

Also note I called it 'object graph' where it was 'command graph'. I dont mind which one we use - which do you prefer? but I do think we should be consistent with that.

The remaining information on the 'Scripting Commands' page seems redundant with information already described (in more verbose form) in the 'Introduction: Object Graph' section. Maybe we should fold that content into the latter?

Lastly for this section, as we have the similarly name 'API commands' section, maybe we should rename the 'Commands API' a bit more generally into something like 'Architecture', and rename what is currently 'API commands' to 'Commands API'.

---

I really like the new graph diagrams for each node. I do wonder if the context would be more clear if each one showed the full graph, but the nodes that are not connected (i.e. currently not shown) are instead faded out/desatured, and likewise any arrows that aren't coming from the node in question. What do you think?

**Comment by elParaguayo** [comment]
Working my way through this but wanted to show you the graphs to see if you like it:
object-graph-root-full dot


object-graph-widget-full dot

File: docs/conf.py
Comment (elParaguayo) [reply docs/conf.py:351]: Yes.

**Comment by elParaguayo** [comment]
> Nice! Is it possible to add the remaining arrows, but in a super light grey?

Yes, should be doable but what do you do about the opposite direction i.e. at the moment, I've just shown widget being able to access bar and screen but screen and bar can also access widgets. I can't do different colours on the same line and I don't want two lines for the same route.

> Now I'm just imagining about the nodes being clickable too...

That's actually pretty easy.

**Comment by m-col** [comment]
> what do you do about the opposite direction i.e. at the moment, I've just shown widget being able to access bar and screen but screen and bar can also access widgets. I can't do different colours on the same line and I don't want two lines for the same route.

I think it'd be clear enough what's going on just having the bidirectional arrow as black.

> That's actually pretty easy.

Cool! Feel free to shoo away ideas like these when I throw them out if you think they are too gimmicky :laughing:

**Comment by m-col** [comment]
Some reported confusions have mentioned various words/phrases used
interchangably, leading to less accessible and useful docs. i.e. easy mistakes
for us close to the code to make (and thus fix).

- Formalise the `Core` as a node in the command graph. and how it relates to
  the backend. We should check docs that mention them and ensure they are
  using them clearly. The backend is a collection of implementations that use backend-specific code (incl. Core, Window, Static, Internal, Drawer), a module or other complete set of implementations for the ABCs in `libqtile/backend/base.py`. It might be neat to provide an absolute minumum backend for use as live documentation and a template upon which new backends could be built, as well as a place to record any kinds of rules they need to play by for the `Qtile` root to interact with it happily.
- Rename the root obj as seen by `qtile cmd-obj` to say _"root":_
  - current: `qtile cmd-obj -o cmd -f windows` (obviously this should continue to
    work for a long time).
  - proposed: `qtile cmd-obj -o root -f windows`

**Comment by elParaguayo** [comment]
>     * Formalise the `Core` as a node in the command graph. and how it relates to
>       the backend. We should check docs that mention them and ensure they are
>       using them clearly. The backend is a collection of implementations that use backend-specific code (incl. Core, Window, Static, Internal, Drawer), a module or other complete set of implementations for the ABCs in `libqtile/backend/base.py`. It might be neat to provide an absolute minumum backend for use as live documentation and a template upon which new backends could be built, as well as a place to record any kinds of rules they need to play by for the `Qtile` root to interact with it happily.

Yes, this is a bit messy. I do find it awkward that the backend's node is called `core` when it's in a `backend` folder and the `core` folder contains the qtile manager!

>     * Rename the root obj as seen by `qtile cmd-obj` to say _"root":_
>       
>       * current: `qtile cmd-obj -o cmd -f windows` (obviously this should continue to
>         work for a long time).
>       * proposed: `qtile cmd-obj -o root -f windows`

Agreed. I would also like it to default to the root node if no object is provided (similar to `lazy`).

I was also thinking that there's no reason that any object on the graph couldn't link back to the root node (its one directional at the moment).

**Comment by elParaguayo** [comment]
Better?
object-graph-widget-grey dot

**Comment by elParaguayo** [comment]
I'm wrapping the code for the graphs in a sphinx directive so, if we make changes, they can be applied to all graphs more quickly.

**Comment by m-col** [comment]
>  I do find it awkward that the backend's node is called core when it's in a backend folder and the core folder contains the qtile manager!

Ha, yeah. The thing with the backend core is that the backend is more than just its `Core`, it's the `Core` and a few other things (like the windows). Hopefully from the POV of users of the graph `libqtile.core` doesn't get confused... though that import path maybe could cause confusion? That said Ive not heard anyone report that yet (besides here).

> Better?

Looks good!

**Comment by elParaguayo** [comment]
Is there anything else we need in this PR or shall I tidy up the commit history so we can merge this?

We do need to make sure commands have a docstrings but that doesn't have to be done here.

**Comment by m-col** [comment]
> Ha, yeah. The thing with the backend core is that the backend is more than just its Core, it's the Core and a few other things (like the windows). Hopefully from the POV of users of the graph libqtile.core doesn't get confused... though that import path maybe could cause confusion? That said Ive not heard anyone report that yet (besides here).

Looks like I repeated myself, I already said this in an earlier comment lol. Gotta stop replying without re-reading the whole thread.

> Is there anything else we need in this PR or shall I tidy up the commit history so we can merge this?

Yeah if you tidy up the commits I'll give it another review with fresh eyes and we can get it merged. Agreed that we don't need to have all docstrings in place before merging - best to keep the momentum than let the PR sit any longer.

**Review by m-col — COMMENTED** [review]
I'm still going through and reviewing but me just say, bravo :clap: :rocket: this is truly fantastic work

File: docs/conf.py
Comment (elParaguayo) [reply docs/conf.py:348]: No

File: docs/qtile_docs/base.py
Comment (elParaguayo) [reply docs/qtile_docs/base.py:48]: Yes

File: docs/index.rst
Comment (elParaguayo) [reply docs/index.rst:25]: :eyes: oops

File: docs/manual/commands/interfaces.rst
Comment (m-col) [inline docs/manual/commands/interfaces.rst:16]: I suspect these are meant to be rendered as bullet points but they are not:
image

File: libqtile/command/base.py
Comment (m-col) [inline libqtile/command/base.py:94]: It looks like this isn't used anywhere?

File: docs/manual/commands/interfaces.rst
Comment (m-col) [inline docs/manual/commands/interfaces.rst:63]: Also here

File: docs/manual/commands/index.rst
Comment (m-col) [inline docs/manual/commands/index.rst:29]: Same here

**Review by m-col — APPROVED** [review]

**Comment by m-col** [comment]
There's a redundant collapse level in the TOC, would it be possible to remove it? I think the outer `Bsp` is the rst header and the inner `Bsp` is the path to the python object. Perhaps there is a way to drop the latter from being part of the sidebar, rather than figuring out a way to stop either from having a link.
image

This is super nice. Would it be possible to centre-align it, and clear the white background so it blends with the RTD background?
image

Do you think this page would be improved by using the graph image from above instead of the list of links? A possible concern would be accessibility, using an image instead of text for navigation, but this isn't something I'm very familiar with.
image

**Comment by elParaguayo** [comment]
Thanks for taking the time to give this a thorough review. I'll work on the changes when I can.

File: docs/manual/commands/interfaces.rst
Comment (elParaguayo) [reply docs/manual/commands/interfaces.rst:63]: If this is referring to bullets, it builds ok for me locally.

File: docs/manual/commands/index.rst
Comment (elParaguayo) [reply docs/manual/commands/index.rst:29]: As above, these bullets build for me locally.

**Comment by elParaguayo** [comment]
> There's a redundant collapse level in the TOC, would it be possible to remove it? I think the outer `Bsp` is the rst header and the inner `Bsp` is the path to the python object. Perhaps there is a way to drop the latter from being part of the sidebar, rather than figuring out a way to stop either from having a link.

This one is a bit of a pain and I'm not sure if I can fix it.

It's caused by `autodoc` which we use to generate the summary. However, the issue is that we are using its automativ linking so we can link from the widget/layout pages to the API page. Creating those links creates the entries in the toctree.

The top title is just from a manually generated header before each object but, if I remove this, I then get an extra level in the toctree with an entry for every command. Ideally I would limit the depth of the toctree on these pages but I can't get it to work.

> This is super nice. Would it be possible to centre-align it, and clear the white background so it blends with the RTD background? 

Done and done.

> Do you think this page would be improved by using the graph image from above instead of the list of links? A possible concern would be accessibility, using an image instead of text for navigation, but this isn't something I'm very familiar with.

I think we should stick with the text link for accessibility reasons. You can click on the nodes in the image but that's a convenience rather than core functionality.

**Pull request merged**
